angular/src/lib/accordion/accordion.component.ts

```diff
@@ -0,0 +1,267 @@
+
+    this.accordionService.changeFocus(this.index);
+
+    if (this.isExpanded) {
+      this.isExpanded = false;
+    } else {
+      if (!this.accordion.multipleVisible) {
+        for (let i = 0; i < this.accordion.tabs.length; i++) {
+          this.accordion.tabs[i].isExpanded = false;
+        }
+      }
+      this.isExpanded = true;
+    }
+    event.preventDefault();
+  }
+
+  findTabIndex() {
+    let index = -1;
+
+    this.accordion.tabs.map((tab, idx) => {
+      if (tab === this) {
+        index = idx;
+      }
+    });
+    return index;
+  }
+
+  getNewIndex(currentIndex, change, lastIdx) {
+
+    const getPossibleIndex = () => {
+      if (currentIndex + change < 0) {
+        return lastIdx;
+      } else if (currentIndex + change > lastIdx) {
+        return 0;
+      }
+      return currentIndex + change;
+    };
+    const possibleIndex = getPossibleIndex();
+
+    if (this.accordion.tabs[possibleIndex].disabled) {
+      return this.getNewIndex(possibleIndex, change, lastIdx);
+    }
+    return possibleIndex;
+  }
+
+  onKeyDown(event: KeyboardEvent) {
+    let newIndex;
+
+    this.handleKeyDown.emit(event.code);
+
+    switch (event.code) {
+      case 'Space':
+      case 'Enter':
+        this.toggle(event);
+
+        break;
+
+      case 'ArrowUp':
+      case 'ArrowLeft':
+        newIndex = this.getNewIndex(this.index, -1, this.accordion.tabs.length - 1);
+
+        this.accordionService.changeFocus(newIndex);
+
+        if (this.accordion.tabs[this.focus]) {
+          this.accordion.tabs[this.focus].headerRef.nativeElement.focus();
+        }
+        break;
+
+      case 'Tab':
+      case 'ArrowRight':
+      case 'ArrowDown':
+        newIndex = this.getNewIndex(this.index, 1, this.accordion.tabs.length - 1);
+
+        this.accordionService.changeFocus(newIndex);
+
+        if (this.accordion.tabs[this.focus]) {
+          this.accordion.tabs[this.focus].headerRef.nativeElement.focus();
+        }
+
+        break;
+
+      case 'PageUp':
+      case 'Home':
+        this.accordionService.changeFocus(0);
+        this.accordion.tabs[0].headerRef.nativeElement.focus();
+
+        break;
+
+      case 'PageDown':
+      case 'End':
+        this.accordionService.changeFocus(this.accordion.tabs.length - 1);
+        this.accordion.tabs[this.accordion.tabs.length - 1].headerRef.nativeElement.focus();
+
+        break;
+
+      default:
+        break;
+    }
+    event.preventDefault();
+  }
+
+  get tabWrapperClasses() {
+    return {
+      ['cui-accordion__group--disabled']: this.disabled,
+      'cui-accordion__group--active': this.isExpanded,
+      [this.tabClass]: this.tabClass
+    };
+  }
+
+  get headerClasses() {
+    return {
+      ['cui-accordion__header--' + this.height]: this.height,
+      'cui-accordion__header--separator': this.showSeparator,
+      [this.headerClass]: this.headerClass
+    };
+  }
+
+  private isHeightValid = () =>
+    ['', '56'].includes(this.height)
```

Comment:
It's good to use includes for more height options. But since we only have 56 and null options, we may have some more effective way to do the validation.

Reply:
removed this check function and moved it to the get headerClasses()